packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -2415,112 +2424,27 @@ export class LitNodeClientNodeJs {
-
-    // -- validate if this.networkPubKeySet is null
-    if (this.networkPubKeySet === null) {
-      return throwError({
-        message: 'networkPubKeySet cannot be null',
-        error: LIT_ERROR.PARAM_NULL_ERROR,
-      });
-    }
-
-    // -- construct SIWE message that will be signed by node to generate an authSig.
-
-    const _expiration =
-      params.expiration ||
-      new Date(Date.now() + 24 * 60 * 60 * 1000).toISOString();
-
-    // Try to get it from local storage, if not generates one~
-    let sessionKey = this.getSessionKey(params.sessionKey);
-    let sessionKeyUri = LIT_SESSION_KEY_URI + sessionKey.publicKey;
-
-    let siweMessage: SiweMessage = new SiweMessage({
-      domain: globalThis.location.host,
-      address: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', // This will be populated by the node.
-      statement: 'Lit Protocol PKP session signature',
-      uri: sessionKeyUri,
-      version: '1',
-      chainId: 1,
-      expirationTime: _expiration,
-      resources: params.resources,
-    });
-
-    let siweMessageStr: string = siweMessage.prepareMessage();
-
-    // ========== Get Node Promises ==========
-    // -- fetch shares from nodes
-    const requestId = this.getRequestId();
-    const nodePromises = this.getNodePromises((url: string) => {
-      return this.getVerifyWebAuthnAuthenticationKeyShare(
-        url,
-        {
-          username: params.username,
-          credential: params.verificationParams,
-          sessionPubkey: sessionKey.publicKey,
-          siweMessage: siweMessageStr,
-        },
-        requestId
-      );
-    });
-
-    // -- resolve promises
-    const res = await this.handleNodePromises(nodePromises);
-
-    // -- case: promises rejected
-    if (!this.#isSuccessNodePromises(res)) {
-      this.throwNodeError(res);
-      return {} as JsonAuthSig;
-    }
-
-    const responseData = res.values;
-    log('responseData', JSON.stringify(responseData, null, 2));
-
-    // ========== Extract shares from response data ==========
-    // -- 1. combine signed data as a list, and get the signatures from it
-    const signedDataList = responseData.map(
-      (r: any) => (r as SignedData).signedData
-    );
-
-    const signatures = this.getSessionSignatures(signedDataList);
-
-    console.log('signatures', signatures);
-
-    const { authSig } = signatures;
-
-    return {
-      sig: authSig.signature,
-      derivedVia: 'webauthn via Lit PKP',
-      signedMessage: authSig.siweMessage,
-      address: computeAddress('0x' + authSig.publicKey),
-    };
-  };
-
-  getVerifyWebAuthnAuthenticationKeyShare = async (
-    url: string,
-    body: GetVerifyWebAuthnAuthenticationKeyShareProps,
-    requestId: string
-  ) => {
-    log('getVerifyWebAuthnAuthenticationKeyShares');
-    const urlWithPath = `${url}/web/auth/webauthn`;
-    return await this.sendCommandToNode({
-      url: urlWithPath,
-      data: body,
-      requestId,
-    });
-  };
+  generateAuthMethodForWebAuthn = (
```

Comment:
May want to consider moving these utility methods outside of this class so one would not have to instantiate the class to use them

Reply:
Future PR 🤞🏻